guides/release/getting-started/core-concepts.md

```diff
@@ -16,49 +17,103 @@ No matter how the URL gets set, the first thing that happens is that the Ember r
 
-Besides properties, double curly braces (`{{}}`) may also contain
-helpers, which we'll discuss later.
+We will go into more details about each of these template features later on in
+this guide.
 
-## Models
+## Components
 
-Models represent persistent state.
+Components allows you to break up your templates and organize them into small,
+self-contained and reusable pieces.
 
-For example, a property rentals application would want to save the details of a rental when a user publishes it, and so a rental would have a model defining its details, perhaps called the _rental_ model.
+In its most basic form, a component is just a piece of template that can be
+referred to by name. Similar to functions in programming languages, they can
+also take _arguments_, allowing them to be customized to the specific context
+they are being rendered into.
 
-A model typically persists information to a web server, although models can be configured to save to anywhere else, such as the browser's Local Storage.
+For example, the example in the previous section is getting a bit long. We can
+_extract_ the snippet for rendering the user's name and profile picture into
+its own component:
 
-## Components
+```handlebars {data-filename="app/components/user-profile.hbs"}
+<img src="{{@user.profileImage}}" alt="{{@user.name}}'s profile picture"> {{@user.name}}
+```
 
-While templates describe how a user interface looks, components control how the user interface _behaves_.
+This allows us to simplify the original template into this:
 
-Components consist of two parts: a template written in Handlebars, and a source file written in JavaScript that defines the component's behavior. For example, our property rental application might have a component for displaying all the rentals called `AllRentals`, and another component for displaying an individual rental called `RentalTile`. The `RentalTile` component might define a behavior that lets the user hide and show the image property of the rental.
+```handlebars {data-filename="app/templates/welcome.hbs"}
+{{!-- The model for this route is the current user --}}
 
-Let's see these core concepts in action by building a property rental application in the next lesson.
+<div>
+  Hi <UserProfile @user={{this.model}} /> this is a valid Ember template!
+</div>
+
+{{#if this.model.isAdmin}}
+  <div>Remember, with great power comes great responsibility!</div>
+{{/if}}
+```
+
+Not only did we cleaned up the original template to be more readable, we now
+have a `<UserProfile>` component that we can reuse whenever we need to render
+information about a given user.
+
+You can think of components as Ember's way for letting you create your own HTML
+tags. In addition to rednering content, components can also have JavaScript
+code associated with them, allowing you to add _behavior_, such as responding
+to a user clicking on your component.
+
+We will cover these advanced component features in a later chapter.
 
 ## Lifecycle Hooks
```

Comment:
This is kind of out-of-scope for my PR, but should this section be removed? The router hook is can be covered in the route handler section above if we want, and the component hook part is largely not applicable in octane anymore (I think?)